### ISSUE TITLE
A sudden change in "validation/main/acc" after loading the snapshot

### ISSUE THREAD

**Issue by mpc001** [espnet/espnet#210]
Hi,

After loading the training model from a snapshot snapshot_iter_136356, it's weird to have 
a sudden change in "validation/main/loss_ctc".

    Before loading the snapshot
![video_acc_0](https://user-images.githubusercontent.com/28733498/41216773-9ffff5e8-6d4d-11e8-9a4d-0f2580898b28.png)

    After loading the snapshot:
![video_acc_2](https://user-images.githubusercontent.com/28733498/41216631-0f013b88-6d4d-11e8-912b-6dcac7fc178c.png)

    Before loading the snapshot
    {
        "validation/main/loss": 51.400978088378906, 
        "main/loss": 128.95794677734375, 
        "main/loss_ctc": 190.42391967773438, 
        "iteration": 136400, 
        "eps": 1e-08, 
        "main/loss_att": 67.49195861816406, 
        "validation/main/loss_att": 25.587093353271484, 
        "elapsed_time": 112193.59979605675, 
        "epoch": 11, 
        "validation/main/acc": 0.7833305597305298, 
        "main/acc": 0.7837930917739868, 
        "validation/main/loss_ctc": 77.21483612060547
    }, 

    After loading the snapshot:
    {
        "main/loss_ctc": 28.216089248657227, 
        "main/loss": 18.790611267089844, 
        "validation/main/loss": 51.408992767333984, 
        "iteration": 136357, 
        "eps": 1e-08, 
        "main/loss_att": 9.365135192871094, 
        "validation/main/loss_att": 25.660505294799805, 
        "elapsed_time": 112159.20889616013, 
        "epoch": 11, 
        "validation/main/acc": 0.7828497886657715, 
        "main/acc": 0.8584474921226501, 
        "validation/main/loss_ctc": 77.15746307373047
    }, 
    {
        "main/loss": 147.9554901123047, 
        "main/loss_ctc": 218.45945739746094, 
        "iteration": 136400, 
        "eps": 1e-08, 
        "main/loss_att": 77.45149230957031, 
        "elapsed_time": 112206.56107521057, 
        "epoch": 11, 
        "main/acc": 0.7835178971290588
    }, 

Thank you very much!

**Comment by sw005320** [comment]
I actually don't test the resume function. @bobchennan, do you have any ideas (I remember that you used that function)?

**Comment by bobchennan** [comment]
Yeah I think resume function didn't work since snapshot didn't include model itself.

This should fix it:
https://github.com/bobchennan/espnet/blob/22ae9cac2cc8f9920631406f9d609897eda78e6b/src/asr/asr_pytorch.py#L226

**Comment by sw005320** [comment]
@bobchennan, can you make a PR about this?

**Comment by mpc001** [comment]
Hi, @bobchennan and @sw005320 

Thanks for your reply.

I implemented in Chainer backend not Pytorch backend. The resume part is as follows. It should be fine.

    # Resume from a snapshot
    if args.resume:
        chainer.serializers.load_npz(args.resume, trainer)

    # set up validation iterator

By the way, the weight might be loaded because "main/acc" seems similar at the iteration 136400. However, I wonder why there is a sudden change.

Thanks in advance!

**Comment by kan-bayashi** [comment]
I'm not sure, but I think, after loading, training started from the middle of the epoch, therefore, the accuracy was calculated using subset of the training data.  Consequently, the accuracy for plotting is overestimated. The fact that the accuracy in epoch 12 is reasonable value support this idea.

**Comment by mpc001** [comment]
Hi, @kan-bayashi,

Thanks for your reply. I replicate the same experiment without reloading during training and get a similar result. I support your idea as well. Thanks for your help!

**Comment by sw005320** [comment]
@mpc001, can you send a PR about your change of the resume part?
